lib/rules/forbid-component-props.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @fileoverview Forbid certain props on components
+ * @author Joe Lencioni
+ */
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Constants
+// ------------------------------------------------------------------------------
+
+var DEFAULTS = ['className', 'style'];
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = function(context) {
```

Comment:
This should use the [new eslint rule format](http://eslint.org/blog/2016/07/eslint-new-rule-format)